### ISSUE TITLE
fix(spherical): every transform returns NaN for grids at T86 and above (Nx ≥ 174, Ny ≥ 87) — `lpmv` overflows before normalisation

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#96]
Part of #81.

## Summary

`SphericalGrid2D` builds its associated-Legendre table by calling `scipy.special.lpmv(m, l, μ)` (unnormalised) and multiplying by `exp(0.5 · log_norm_sq)` afterwards. Unnormalised `P_l^m` grows like `(2m−1)!!`, and `lpmv` returns `inf` at `(m, l) = (85, 86)`. Any grid whose FFT wavenumber range reaches `m = 85` and whose degree range reaches `l = 86` therefore has `inf`/`nan` in `_P_lm`, and the NaN column poisons every transform output through the longitude IFFT. On the standard `Nx = 2Ny` grid that is T86 and above. The module docstring (`grid.py:24-28`) claims stability "up to about l ≈ 512"; it is stable only to T85. No warning or error is raised.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from spectraldiffx import SphericalGrid2D
for Nx, Ny in ((172, 86), (174, 87), (256, 128), (512, 256)):
    g = SphericalGrid2D.from_N_L(Nx=Nx, Ny=Ny, Lx=2*np.pi, Ly=np.pi)
    u = jnp.cos(g.y)[:, None] * jnp.ones((1, Nx))
    print(Nx, Ny, "transform finite?", bool(jnp.isfinite(g.transform(u)).all()))
```

```
172 86  transform finite? True     (T85)
174 87  transform finite? False    (T86, smallest broken standard grid)
256 128 transform finite? False    (T127: 241920 / 4194304 table entries non-finite)
512 256 transform finite? False    (T255: 7528960 / 33554432 non-finite)
```

Raw `lpmv` at the threshold: `(m=84, l=99)` finite, max 1.6e161; `(85, 85)` finite, 6.6e152; `(85, 86)` inf. Grids with `Nx = 64, Ny = 512` are finite (m limited by Nx) and `Nx = 1024, Ny = 32` are finite (rows `l < |m|` are zero), so the failure is specifically the `(m ≥ 85, l ≥ 86)` corner.

## Root cause

`spectraldiffx/_src/spherical/grid.py:104-143` (`_alp_matrix`) and `:497-502` (2D table assembly): normalisation applied after an already-overflowed value.

## Proposed fix

Compute **normalised** associated Legendre functions directly with the standard stable recurrence per order `m`:

```
P̃_m^m(μ)     = (−1)^m sqrt((2m+1)/2 · (2m)! / (4^m (m!)²)) · (1−μ²)^{m/2}     (evaluate in log space)
P̃_{m+1}^m(μ) = sqrt(2m+3) · μ · P̃_m^m(μ)
P̃_l^m(μ)     = a_lm (μ P̃_{l−1}^m − b_lm P̃_{l−2}^m),
                a_lm = sqrt((4l²−1)/(l²−m²)),  b_lm = sqrt(((l−1)²−m²)/(4(l−1)²−1))
```

with `P̃` orthonormal on `[−1, 1]` w.r.t. `dμ`, which is the convention the current table uses (verified: Gram diagonal exactly `2π/Nx²`, off-diagonal 8e-17). The audit checked this recurrence against the lpmv-based values for `m ≤ 50`: max difference 3.5e-13; and it is finite at `m = 150`.

Implement it in numpy at construction (drop-in), or in JAX with `lax.scan` over `l` and `vmap` over `m`, which additionally makes grid construction traceable (needed by the "GFD solvers cannot jit" issue in this epic) and removes the 36 s Python-loop build at T255.

## Acceptance criteria

- [ ] Test: `SphericalGrid2D(Nx=256, Ny=128)` table is finite; forward/inverse round trip of a band-limited field at T127 is < 1e-10.
- [ ] Test: new table matches the old one to 1e-12 for `Nx=64, Ny=32` (regression on the convention).
- [ ] Test: table entries for `(l, m) = (200, 150)` at `Ny = 256` match `scipy.special.sph_harm_y`-derived values (up to the documented scale) to 1e-10.
- [ ] Docstring stability claim corrected; construction time at T255 reported in the PR.
